satpy/readers/lsasaf_h5.py

```diff
@@ -0,0 +1,231 @@
+
+        return variable
+
+    @staticmethod
+    def _mask_variable(variable):
+        if 'MISSING_VALUE' in variable.attrs:
+            variable = variable.where(
+                variable != variable.attrs['MISSING_VALUE'])
+            variable.attrs['MISSING_VALUE'] = np.nan
+        return variable
+
+    def _get_proj_lon(self):
+        """Find projection longitude from the projection name info.
+
+        LSA SAF data doesn't contain the projected subsatellite longitude, only the actual satellite longitude.
+        So we have to find the projection longitude from the name of the projection given in the attributes.
+        """
+        proj_name = self.nc.attrs['PROJECTION_NAME']
+        pos_s = proj_name.find('<')
+        pos_e = proj_name.find('>')
+
+        try:
+            proj_lon = float(proj_name[pos_s + 1:pos_e])
+        except ValueError:
+            raise ValueError(f'Could not find projection longitude from the projection name: {proj_name}')
+
+        return proj_lon
+
+    def get_area_def(self, dsid):
+        """Get the area definition of the datasets in the file.
+
+        Only applicable for MSG products!
+        """
+        pdict = {}
+        pdict['coff'] = self.nc.attrs['COFF']
+        pdict['loff'] = -self.nc.attrs['LOFF'] + 1
+        pdict['cfac'] = self.nc.attrs['CFAC']
+        pdict['lfac'] = self.nc.attrs['LFAC']
+
+        # Unfortunately this dataset does not store a, b or h.
+        # We assume a, rf and h here based on EUMETSAT standard values.
+        pdict['a'] = 6378169
+        pdict['b'] = 6356583.8
+        pdict['h'] = 35785831.0
+
+        pdict['ssp_lon'] = self._get_proj_lon()
+        pdict['nlines'] = float(self.nc.attrs['NL'])
+        pdict['ncols'] = float(self.nc.attrs['NC'])
+
+        pdict['scandir'] = 'S2N'
+        pdict['a_desc'] = 'MSG/SEVIRI low resolution channel area'
+        pdict['a_name'] = 'geosmsg'
+        pdict['p_id'] = 'msg_lowres'
+
+        area_extent = get_area_extent(pdict)
+        area_extent = (area_extent[0],
+                       area_extent[1],
+                       area_extent[2],
+                       area_extent[3])
+
+        area = get_area_definition(pdict, area_extent)
+
+        return area
+
+    @staticmethod
+    def _ensure_crs_extents_in_meters(crs, area_extent):
+        """Fix units in Earth shape, satellite altitude and 'units' attribute."""
+        if 'kilo' in crs.axis_info[0].unit_name:
+            proj_dict = crs.to_dict()
+            proj_dict["units"] = "m"
+            if "a" in proj_dict:
+                proj_dict["a"] *= 1000.
+            if "b" in proj_dict:
+                proj_dict["b"] *= 1000.
+            if "R" in proj_dict:
+                proj_dict["R"] *= 1000.
+            proj_dict["h"] *= 1000.
+            area_extent = tuple([val * 1000. for val in area_extent])
+            crs = CRS.from_dict(proj_dict)
+        return crs, area_extent
```

Comment:
Unused method? If not, I'm tempted to say we should just deal with the projection in kilometers if possible.